### PR TITLE
Minor optimization for Signup and Login process.

### PR DESCRIPTION
**Pull request by huxuan**
- Use `exit(err_msg($err_code))` rather than `echo`.
- Change the logical order of different error type and exit once a more important one occurred.
<del>- Redirect to login page after signup successfully. (This is more reasonable than index page, I think)</del>
- Introduce custom email library to make send emails easier in controller.
- Other minor change, like add an empty line between different code 'sections'.

### CODE REVIEW

File: application/views/signup_form.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <script type="text/javascript">
     var controller = "<?=site_url('user/signup')?>";
-    var directto = "<?=base_url()?>";
+    var directto = "<?=site_url('user/login')?>";
```
Comment (JeldorPKU) [inline application/views/signup_form.php]: The users are not supposed to login immediately after they signed up. The administrator will confirm the user after which the user can login.

Comment (huxuan) [inline application/views/signup_form.php]: OK, this is not the point. I have revert the change here.
